dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/dataintegrity/DefaultDataIntegrityService.java

```diff
@@ -517,22 +519,36 @@ private void registerNonDatabaseIntegrityCheck( DataIntegrityCheckType type,
         Supplier<List<DataIntegrityIssue>> check )
     {
         String name = type.getName();
-        checksByName.put( name, DataIntegrityCheck.builder()
-            .name( name )
-            .severity( DataIntegritySeverity.WARNING )
-            .section( "Legacy" )
-            .description( name.replace( '_', ' ' ) )
-            .runDetailsCheck( c -> new DataIntegrityDetails( c, new Date(), null, check.get() ) )
-            .runSummaryCheck( c -> new DataIntegritySummary( c, new Date(), null, check.get().size(), null ) )
-            .build() );
+        I18n i18n = i18nManager.getI18n( DataIntegrityService.class );
+        BinaryOperator<String> info = ( property, defaultValue ) -> i18n
+            .getString( format( "data_integrity.%s.%s", name, property ), defaultValue );
+
+        try
+        {
+            checksByName.put( name, DataIntegrityCheck.builder()
+                .name( name )
+                .displayName( info.apply( "name", name.replace( '_', ' ' ) ) )
+                .severity( DataIntegritySeverity.valueOf(
+                    info.apply( "severity", DataIntegritySeverity.WARNING.name() ).toUpperCase() ) )
+                .section( info.apply( "section", "Other" ) )
+                .description( info.apply( "description", null ) )
+                .introduction( info.apply( "introduction", null ) )
+                .recommendation( info.apply( "recommendation", null ) )
+                .runDetailsCheck( c -> new DataIntegrityDetails( c, new Date(), null, check.get() ) )
+                .runSummaryCheck( c -> new DataIntegritySummary( c, new Date(), null, check.get().size(), null ) )
+                .build() );
+        }
+        catch ( Exception ex )
+        {
+            log.error( "Failed to register data integrity check " + type, ex );
+        }
     }
 
     /**
      * Maps all "hard coded" checks to their implementation method and registers
      * a {@link DataIntegrityCheck} to perform the method as
      * {@link DataIntegrityDetails}.
      */
-    @PostConstruct
```

Comment:
FYI: I needed to defer this further as now when i18n is used this would need setting which would need hibernate which would fail at the stage this needed to happen.